### ISSUE TITLE
How to show boundary map

### ISSUE THREAD

**Issue by a462428** [protossw512/AdaptiveWingLoss#11]
Hi, really thx for you excellent contribution.
I wonder how to show boundary map in your code  (boundary_channels) like the boundary heat map show in your paper

Thanks!

**Comment by protossw512** [comment]
If you look at the heatmap prediction output, you will find it has N+1 channels where N is the number of landmarks. The last channel is the boundary map you need. Then you may apply a colormap to get the desired visualization in my paper.

**Comment by a462428** [comment]
Thx!
However, the response of boundary heatmap is very low. Is that normal?

Heat_bound = outputs[-1][:,-1,:,:].detach().cpu().numpy()
Boundary = Heat_bound[0]
Boundary = np.uint8(Boundary)
Boundary = cv2.applyColorMap(boundary, cv2.COLORMAP_RAINBOW)

**Comment by protossw512** [comment]
@a462428 The value of boundary map are floating numbers from 0 to 1, so you have to multiply it by 255 before converting it to uint8.

Boundary = np.uint8(Boundary * 255)

Hope that helps
